ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/trino/sampler.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Helper module to handle data sampling
+for the profiler
+"""
+from sqlalchemy import and_, inspect, text
+
+from metadata.generated.schema.entity.data.table import TableData
+from metadata.profiler.orm.registry import FLOAT_SET
+from metadata.profiler.processor.handle_partition import RANDOM_LABEL
+from metadata.profiler.processor.sampler.sqlalchemy.sampler import SQASampler
+
+
+class TrinoSampler(SQASampler):
+    """
+    Generates a sample of the data to not
+    run the query in the whole table.
+    """
+
+    def fetch_sample_data(self) -> TableData:
+        """
+        Use the sampler to retrieve sample data rows as per limit given by user
+        :return: TableData to be added to the Table Entity
+        """
+        if self._profile_sample_query:
+            return self._fetch_sample_data_from_user_query()
+
+        # Add new RandomNumFn column
+        rnd = self.get_sample_query()
+        sqa_columns = [col for col in inspect(rnd).c if col.name != RANDOM_LABEL]
+
+        sqa_sample = (
+            self.client.query(*sqa_columns)
+            .select_from(rnd)
+            .where(
+                and_(
```

Comment:
that should probably be an `or`

Reply:
the `and_` condition would make sure that the `col1` `col2` and `col3` have is_nan as False making sure all 3 columns have no NaNs, even if one column comes in with NaN, the sample data ingestion will fail

Reply:
if you do the `and_` if `col1` has nan, but `col2` and `col3` don't it will not filter the row. That's why we should have `or_` so that if either of the column has nan it will filter that row